src/autopas/containers/cellPairTraversals/CSlicedBasedTraversal.h

```diff
@@ -0,0 +1,133 @@
+ * @date 24 May 2020
+ * @author fischerv
+ */
+
+#pragma once
+
+#include <algorithm>
+
+#include "autopas/containers/cellPairTraversals/SlicedBasedTraversal.h"
+#include "autopas/utils/DataLayoutConverter.h"
+#include "autopas/utils/ThreeDimensionalMapping.h"
+#include "autopas/utils/WrapOpenMP.h"
+
+namespace autopas {
+
+/**
+ * This class provides the colored sliced traversal.
+ *
+ * The traversal finds the longest dimension of the simulation domain and cuts
+ * the domain into as many slices as possible along this dimension. Unlike the regular
+ * sliced traversal, this version uses a 2-coloring to prevent race conditions, instead of
+ * locking the starting layers. This could also be describes as a c02-traversal. This class
+ * is however not derived from CBasedTraversal, as that would not allow varying slice thicknesses,
+ * and would prevent us from selecting the dimension in which we cut the slices.
+ *
+ * @tparam ParticleCell The type of cells.
+ * @tparam PairwiseFunctor The functor that defines the interaction of two particles.
+ * @tparam dataLayout
+ * @tparam useNewton3
+ */
+template <class ParticleCell, class PairwiseFunctor, DataLayoutOption::Value dataLayout, bool useNewton3>
+class CSlicedBasedTraversal : public SlicedBasedTraversal<ParticleCell, PairwiseFunctor, dataLayout, useNewton3> {
+ public:
+  /**
+   * Constructor of the colored sliced traversal.
+   * @param dims The dimensions of the cellblock, i.e. the number of cells in x,
+   * y and z direction.
+   * @param pairwiseFunctor The functor that defines the interaction of two particles.
+   * @param interactionLength Interaction length (cutoff + skin).
+   * @param cellLength cell length.
+   */
+  explicit CSlicedBasedTraversal(const std::array<unsigned long, 3> &dims, PairwiseFunctor *pairwiseFunctor,
+                                 const double interactionLength, const std::array<double, 3> &cellLength)
+      : SlicedBasedTraversal<ParticleCell, PairwiseFunctor, dataLayout, useNewton3>(dims, pairwiseFunctor,
+                                                                                    interactionLength, cellLength) {}
+
+  /**
+   * The main traversal of the colored sliced traversal.
+   * This provides the structure of the loops and its parallelization.
+   *
+   * @copydetails C01BasedTraversal::c01Traversal()
+   *
+   * @tparam allCells Defines whether or not to iterate over all cells with the loop body given as argument. By default
+   * (allCells=false) it will not iterate over all cells and instead skip the last few cells, because they will be
+   * covered by the base step. If you plan to use the default base step of the traversal on this function, use
+   * allCells=false. If you plan to just iterate over all cells, e.g., to iterate over verlet lists saved within the
+   * cells, use allCells=true. For the sliced step if allCells is false, iteration will not occur over the last layer of
+   * cells (for _overlap=1) (in x, y and z direction).
```

Comment:
```suggestion
   * @tparam allCells Defines whether or not to iterate over all cells with the loop body given as argument.
   * If set to false, loopBody will not be applied to the last layer of cells (times overlap) in each dimension.
   * This is sufficient for base steps that only cover spacially forward neighbors, like c08.
   * For base steps that only cover index-forward neighbors like c18, it is necessary to also apply the base
   * step to the last cells, because it calculates interactions with cells spacially behind the base cell.
```